### DISCUSSION TITLE
ReadonlyArray<TData>

### DISCUSSION THREAD

**Discussion by homburg** [TanStack/table#4744] [Ideas]
I'd like to convert all types like `TData[]` to `ReadonlyArray<TData>` (my preference) or `readonly TData[]`.

This would be a non-breaking change as `TData[]` extends/is assignable to `ReadonlyArray<TData>[]`.

This would enable using readonly data for table data and signal explicitly that tanstack-table does not mutate the original data internally anywhere.

**Comment by MH4GF** [comment]
I too wish this request. related: https://github.com/TanStack/table/issues/3748

**Comment by vdawg-git** [comment]
Bumping this. readonly is just nicer
